opentelemetry-sdk/src/trace/config.rs

```diff
@@ -10,12 +10,14 @@ use std::env;
 use std::str::FromStr;
 
 /// Default trace configuration
+#[deprecated(since = "0.23.0", note = "Use Config::default() instead")]
```

Comment:
As shared in https://github.com/open-telemetry/opentelemetry-rust/issues/1810, and https://github.com/open-telemetry/opentelemetry-rust/pull/1788#issuecomment-2120868947 I am not seeing strong value in this `Config` struct abstraction, so this might change further depending on how that issue is going to be resolved.

Okay to proceed with this PR.

Reply:
Don't think it's conflicting even if we want to remove the Config. I think we can merge as it is and iterate on it